### PR TITLE
Fix/days notice warning

### PR DESCRIPTION
**Pull request by paulmatchett**
Notice warning Message not displaying correctly when required. Fixes for this included:

- DaysDuration needed to be passed down as a prop to bookingModal. This value is not required initially, but was given a default value of 1.
- Message not displaying when re selecting annual holiday. eventTypeId was being checked against a string instead of an number.

### CODE REVIEW

File: web-app/src/components/BookingModal/BookingModalForm/container.js [hunk 1]
```diff
@@ -67,7 +72,10 @@ const Container = Wrapped =>
         }
       } else if (name === 'isHalfday' && formData.isHalfday) {
         formData.end = formData.start;
-        formData.eventTypeId = 1;
+        const annualLeave = 1;
+        formData.eventTypeId = annualLeave;
```
Comment (William-Unosquare) [inline web-app/src/components/BookingModal/BookingModalForm/container.js]: Would it not be better to use `formData.eventTypeId = eventTypes.ANNUAL_LEAVE` here?

File: web-app/src/components/BookingModal/BookingModalForm/index.js [hunk 2]
```diff
@@ -42,13 +42,14 @@ const BookingModalForm = props => {
 
   const composeErrorMessage = () => {
     const { eventTypeId, start } = formData;
-    if (isEventBeingUpdated || eventTypeId !== 1) {
+    const annualLeave = 1;
+    if (isEventBeingUpdated || eventTypeId !== annualLeave) {
```
Comment (BrownMark) [inline web-app/src/components/BookingModal/BookingModalForm/index.js]: Same as William's comment above, we have constants for these

File: web-app/src/components/BookingModal/BookingModalForm/container.js (hunk 1 above)
Comment (paulmatchett) [reply web-app/src/components/BookingModal/BookingModalForm/container.js]: @William-Unosquare - Done. Also changed HOLIDAY to ANNUAL_LEAVE in eventTypes and updated files where neccessary. See updated commits below

File: web-app/src/components/BookingModal/BookingModalForm/index.js (hunk 2 above)
Comment (paulmatchett) [reply web-app/src/components/BookingModal/BookingModalForm/index.js]: @BrownMark  - rookie mistake. this has been updated

**Review by William-Unosquare — APPROVED** [review]

**Pull request merged**